Adafruit_DAP.h

```diff
@@ -205,91 +211,87 @@ class Adafruit_DAP {
 // DAP for SAM
 class Adafruit_DAP_SAM : public Adafruit_DAP {
 public:
-  Adafruit_DAP_SAM(void){};
+  Adafruit_DAP_SAM(void) : locked(false) {};
   ~Adafruit_DAP_SAM(void){};
 
+  static const size_t PAGESIZE = SAM_PAGE_SIZE;
+  static const size_t USER_ROW_SIZE = 256;
   static device_t devices[];
-  device_t target_device;
+  bool locked;
 
   bool select(uint32_t *id);
+  virtual void resetWithExtension(void);
+  virtual void finishReset(void);
   void deselect(void);
   void erase(void);
   void lock(void);
-  void programBlock(uint32_t addr, const uint8_t *buf, uint16_t size = 256);
-  void readBlock(uint32_t addr, uint8_t *buf);
+  virtual size_t pageSize() { return PAGESIZE; }
+  void programFlash(const uint8_t * source, uint16_t bootloaderSize, bool doVerify = false, uint32_t flashOffset = 0);
```

Comment:
[Mandatory] `programFlash()` API should not be a part of DAP interface, it is OK for similar progtamming sketch to have similar flow control, the DAP layer shouldn't be concerned with bootloader size.

Reply:
I certainly see the arguments for that view, but on the other hand there really is no very hard limit between the layers. The Adafruit_DAP class having a good sample of how to flash and verify the flash didn't seem to hurt. This certainly isn't important for me to have in the library.

Reply:
Ah I see, this is actually great to have as well, I was misunderstanding the name of `bootloaderSize` arguments. Thinking you have to do some special handling with bootloader. Though looking at the implementation more closely, it is more like a  generic flashing data. Would you mind changing it to more generic 

`bool programFlash(uint32_t addr, const uint8_t *buf, uint32_t count, bool verify = true)`